code/game/objects/structures/tank_dispenser.dm

```diff
@@ -95,24 +95,26 @@
 		if(href_list["oxygen"])
 			if(oxygentanks > 0)
 				var/obj/item/tank/oxygen/O
-				if(oxytanks.len == oxygentanks)
-					O = oxytanks[1]
+				if(oxytanks.len) // If an oxygen tank was ever added by a player, this will be >= 1
+					O = oxytanks[1] // Use the already generated oxy tank so we don't have to make a new one
 					oxytanks.Remove(O)
 				else
-					O = new /obj/item/tank/oxygen(loc)
-				O.loc = loc
+					O = new /obj/item/tank/oxygen

```

Comment:
Not your fault, but this code is really gross.

IT should probably be refactored one day so the tanks are just pre-spawned inside these structures and you take them out/remove them.

Creating them, on demand and basing it around.....a number is just *weird*.

Reply:
I thought so too. I bet I could do it fairly easily, might as well since I'm here.